src/kit.ts

```diff
@@ -174,22 +176,44 @@ export async function kitIfCompiler(bin: string, pr?: ProgressReporter): Promise
     }
     const name = `GCC ${description}${version}`;
     log.debug('Detected GCC compiler:', bin);
+    let gccKit: Kit = {
+      name,
+      compilers: {
+        C: bin,
+      }
+    };
+
     if (await fs.exists(gxx_bin)) {
-      return {
-        name,
-        compilers: {
-          CXX: gxx_bin,
-          C: bin,
-        }
-      };
-    } else {
-      return {
-        name,
-        compilers: {
-          C: bin,
+      gccKit = {name, compilers: {C: bin, CXX: gxx_bin}};
+    }
+
+    const isWin32 = process.platform === 'win32';
+    if (isWin32 && bin.toLowerCase().includes('mingw')) {
+      const binParentPath = path.dirname(bin);
+      const mingwMakePath = path.join(binParentPath, 'mingw32-make.exe');
+      if (await fs.exists(mingwMakePath)) {
+
+        // Check for working mingw32-make
+        const execMake = await proc.execute(mingwMakePath, ['-v'], null, {environment: {PATH: binParentPath}}).result;
+        if (execMake.retc !== 0) {
+          log.debug('Bad mingw32-make binary ("-v" returns non-zero):', bin);
+        } else {
+          let make_version_output = execMake.stdout;
+          if (make_version_output.length == 0)
+            make_version_output = execMake.stderr;
+          const output_line_sep = make_version_output.trim().split('\n');
+          const isMake = output_line_sep[0].includes('Make');
+          const isMingwTool = output_line_sep[1].includes('mingw32');
+
+          if (isMake && isMingwTool) {
+            gccKit.preferredGenerator = {name: 'MinGW Makefiles'};
+            gccKit.environmentVariables = {PATH: binParentPath};
```

Comment:
We'll still want to append to the regular `%PATH%`. This will just replace it with the single MinGW path and create a lot of trouble.